Source/CarthageKit/Project.swift

```diff
@@ -542,31 +461,28 @@ public final class Project {
 						return self.createVersionFilesForFrameworks(frameworkURLs, fromDirectoryURL: directoryURL, projectName: projectName, commitish: commitish)
 					}
 					.then(SignalProducer<URL, CarthageError>(value: directoryURL))
-		}
+			}
 	}
 
 	/// Removes the file located at the given URL
 	///
 	/// Sends empty value on successful removal
 	private func removeItem(at url: URL) -> SignalProducer<(), CarthageError> {
-		return SignalProducer<URL, CarthageError>(value: url)
-			.attemptMap({ (url: URL) -> Result<(), CarthageError> in
-				do {
-					try FileManager.default.removeItem(at: url)
-					return .success()
-				} catch let error as NSError {
-					return .failure(.writeFailed(url, error))
-				}
-			})
```

Comment:
Wanted to check for type-inference before making suggestions here, but removing the annotation on `attemptMap` just compiled for me, so in the style of [`func cacheDownloadedBinary`](https://github.com/Dschee/Carthage/blame/16f043a39840b6514cf90d3130ab9ff98cecb227/Source/CarthageKit/Project.swift#L1052-L1060), we should do:
```swift
	private func removeItem(at url: URL) -> SignalProducer<(), CarthageError> {
		return SignalProducer(value: url)
			.attemptMap { url in
				do {
					try FileManager.default.removeItem(at: url)
					return .success(())
				} catch let error as NSError {
					return .failure(.writeFailed(url, error))
				}
			}
	}
```